### PR TITLE
Fix for Txmake options in RIB.

### LINKED ISSUE #198: Texture parameter errors
Minor ignorable error when rendering, but I noticed there was a "int filename" parameter put into the rib stream when using a PxrTexture node.  It doesn't cause a problem, but there's already of course a "string filename" parameter to PxrTexture so it throws an error like this:

PRMan: R10043 {WARNING} filename, unknown or mismatched input parameter of PxrTexture (PxrTexture)".

Not sure where the other parameter is get getting in there?  Maybe from the txmake options?

### PR DESCRIPTION
**Pull request by MWY3510**
Sorry there is a lot of RIB archive code in here also. Most is still elsewhere. All this merge does do is fix #198 .

### CODE REVIEW

**Comment by bsavery** [comment]
Ok wow I'll have to review all that

On Tuesday, February 16, 2016, MWY3510 <notifications@github.com> wrote:

> Sorry there is a lot of RIB archive code in here also. Most is still
> elsewhere. All this merge does do is fix #198
> <https://github.com/bsavery/PRMan-for-Blender/issues/198> .
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/bsavery/PRMan-for-Blender/pull/199
> Commit Summary
>
>    - UI changes for RIB archive
>    - Improvements to RIB archive UI
>    - Quick and dirty fix for strands
>    - More RIB archive items
>    - Fix for stands.
>    - Bring up to date for merge corrections
>    - bsavery's changes
>    - Remove double and add some scene properties
>    - Merge pull request #47 from bsavery/master
>    - More RIB archiving
>    - Export For Archives Done.
>    - Merge pull request #48 from bsavery/master
>    - More archiving stuff
>    - Basic mesh import from archives
>    - Merge pull request #49 from bsavery/master
>    - Merge pull request #50 from bsavery/master
>    - Manifest File part one.
>    - Merge branch 'master' of
>    https://github.com/MWY3510/PRMan-for-Blender.git
>    - Merge branch 'master' of https://github.com/bsavery/PRMan-for-Blender
>    - Remove unused function.
>    - Remove uneeded UI
>    - Finish archive export and manifest definition.
>    - Merge pull request #52 from bsavery/master
>    - Remove prints and Fix txmake options appearing in RIB
>    - Disable RIB archiving for the moment.
>
> File Changes
>
>    - *M* export.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-0>
>    (145)
>    - *M* nodes.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-1>
>    (5)
>    - *M* operators.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-2>
>    (64)
>    - *M* properties.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-3>
>    (24)
>    - *M* ui.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-4>
>    (33)
>    - *M* util.py
>    <https://github.com/bsavery/PRMan-for-Blender/pull/199/files#diff-5>
>    (71)
>
> Patch Links:
>
>    - https://github.com/bsavery/PRMan-for-Blender/pull/199.patch
>    - https://github.com/bsavery/PRMan-for-Blender/pull/199.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/199>.
>


-- 
brian.savery@gmail.com
508-274-8700

**Comment by MWY3510** [comment]
Yea, it looks worse than it is. Most of this is related to RIB archiving which the last commit effectively disables. Let me know if you have any questions.

**Comment by bsavery** [comment]
How close is the rib archiving?  I have some free time over the next few
days if I can finish it

On Tuesday, February 16, 2016, MWY3510 <notifications@github.com> wrote:

> Yea, it looks worse than it is. Most of this is related to RIB archiving
> which the last commit effectively disables. Let me know if you have any
> questions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/199#issuecomment-184985498>
> .
>


-- 
brian.savery@gmail.com
508-274-8700

**Comment by MWY3510** [comment]
Sorry for the late reply Brian. At the time of this writing the RIB archiving is about 75% done. Currently you can export and use any archive as a single frame object (no animations.) This works for almost everything, the only bug I have is particle systems. Since they are global objects they will remain where they exported at and not at the new object. Some math could fix that but besides that everything that can be rendered normally can be imported as an archive even multi-material objects.



What is still left to be done is the animation part. The manifest system and export portion of RIB archives already produce animated archives but I have not finished the import of these archives.

Of course all of this is implemented in a very messy manor (from the standpoint of the code not stability.) 

Let me know if you have any questions. I will be plugging away on this during the next few days also (making up for lost time hopefully.) If you want to help it is appreciated. What is in the 1.0 repository is my latest.

Ryan

**Comment by MWY3510** [comment]
Also, If you want to have a more in depth conversation on this then I am available after 2:30PM my time tomorrow the 17th. (11:30AM your time I think.)

Ryan

**Comment by MWY3510** [comment]
Also here is a picture. All of the objects are being pulled from archives Located in different locations. And much to GitHub's relief I am done for tonight. Happy Rendering!

archives

File: export.py [hunk 1]
```diff
@@ -305,7 +308,8 @@ def get_strands(scene, ob, psys):
     scalpT = []
     nverts = 0
     for pindex in range(total_hair_count):
-        if not psys.settings.show_guide_hairs and pindex < num_parents:
+        
+        if pindex < num_parents:    # Removed display guide hairs button
```
Comment (bsavery) [inline export.py]: Why was this removed?

File: export.py [hunk 2]
```diff
@@ -1874,6 +1879,7 @@ def export_data_archives(ri, scene, rpass, data_blocks):
     for name, db in data_blocks.items():
         if not db.do_export:
             continue
+        print("Exporting archive: " + db.archive_filename)
```
Comment (bsavery) [inline export.py]: Debug print?

**Comment by bsavery** [comment]
Two questions.  1.  Don't see the fix for texture filename property in there.  Am I missing it?

2.  So what is the overall goal of the manifest file part in archives.  Why is it needed?

**Comment by MWY3510** [comment]
Ok, Just addressing the pull request. Thanks for catching the print I will correct that. Second, the removal of show_guide_hairs was purely a mistake on my part. I do have a fix for texture filename property. The strange thing is that what I did should not have been necessary?

**Comment by MWY3510** [comment]
Ok, I am going to close this for the moment. I will be "fixing" my git repositories to be much cleaner. This will involve possibly deleting both my public and private forks. I should have this resolved by the end of tomorrow. In addition this will also let me merge the archiving changes in an easier to understand way. Sorry for the inconvenience.

**Comment by bsavery** [comment]
No worries.  The other thought I had was checking the 1.0ish features into
the devel branch...

On Mon, Feb 22, 2016 at 9:02 PM, MWY3510 <notifications@github.com> wrote:

> Closed #199 <https://github.com/bsavery/PRMan-for-Blender/pull/199>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bsavery/PRMan-for-Blender/pull/199#event-560991244>.
>



-- 
brian.savery@gmail.com
508-274-8700

**Comment by MWY3510** [comment]
Great, I was going to clean out all of the left over code (mostly my fault.) I guess I will attempt to merge with the devel branch next time. That way you should have less work reviewing redundant code :) .
